lhotse/recipes/fisher_english.py

```diff
@@ -0,0 +1,177 @@
+        for audio_partition_dir in audio_dir_path.iterdir():
+            audio_partition_dir_path = audio_dir_path / audio_partition_dir / 'audio'
+            audio_subdir_paths += [audio_partition_dir_path / audio_subdir 
+                                   for audio_subdir in audio_partition_dir_path.iterdir()]
+            
+    transcript_subdir_paths = []
+    for transcript_dir in transcript_dirs:
+        transcript_dir_path = corpus_path / transcript_dir / 'data' / 'trans'
+        transcript_subdir_paths += [transcript_dir_path / transcript_subdir 
+                                    for transcript_subdir in transcript_dir_path.iterdir()]
+    
+    audio_paths = walk_dirs_parallel(audio_subdir_paths, '*.sph', 'Parsing audio sub-dirs')
+    transcript_paths = walk_dirs_parallel(transcript_subdir_paths, '*.txt', 'Parsing transcript sub-dirs')
+    
+    sessions = {}
+    for transcript_dir in transcript_dirs:
+        sessions_data_path = check_and_rglob(corpus_path / transcript_dir / 'doc', '*_calldata.tbl')[0]
+        with codecs.open(sessions_data_path, 'r', 'utf8') as sessions_data_f:
+            tmp_sessions = [l.rstrip('\n').split(',') for l in sessions_data_f.readlines()][1:]
+            sessions.update({l[0]: {'A': l[5], 'B': l[10]} for l in tmp_sessions})
+
+    assert len(transcript_paths) == len(sessions) == len(audio_paths)
+    
+    create_recordings_input = [(p, None if absolute_paths else 5) for p in audio_paths]
+    recordings = [None] * len(audio_paths)
+    with ThreadPoolExecutor(os.cpu_count() * 4) as executor:
+        with tqdm(total=len(audio_paths)) as pbar:
+            pbar.set_description('Collect recordings')
+            for i, reco in enumerate(executor.map(create_recording, create_recordings_input)):
+                recordings[i] = reco
+                pbar.update()
+        
+    recordings = RecordingSet.from_recordings(recordings)
+    
+    create_supervisions_input = [(sessions, p) for p in transcript_paths]
+    supervisions = [None] * len(create_supervisions_input)
+    with ThreadPoolExecutor(os.cpu_count() * 4) as executor:
+        with tqdm(total=len(create_supervisions_input)) as pbar:
+            pbar.set_description('Create supervisions')
+            for i, tmp_supervisions in enumerate(executor.map(create_supervision, create_supervisions_input)):
+                supervisions[i] = tmp_supervisions
+                pbar.update()
+    supervisions = list(it.chain.from_iterable(supervisions))
+    supervisions = SupervisionSet.from_segments(supervisions)
+    
+    supervisions = trim_supervisions_to_recordings(recordings, supervisions)
+    validate_recordings_and_supervisions(recordings, supervisions)
+
+    if output_dir is not None:
+        output_dir = Path(output_dir)
+        output_dir.mkdir(parents=True, exist_ok=True)
+        recordings.to_json(output_dir / 'recordings.json')
+        supervisions.to_json(output_dir / 'supervisions.json')
```

Comment:
Since Fisher is large at ~2000h, I suggest writing both of those as `jsonl` instead, like this:
```suggestion
        supervisions.to_file(output_dir / 'supervisions.jsonl.gz')
```

It will enable others to use the lazy manifest mechanism if needed.

Reply:
ok, will change now.